models/contacts.js

```diff
@@ -1,14 +1,74 @@
-// const fs = require('fs/promises')
+// const addContact = async (body) => {}
 
-const listContacts = async () => {}
+const fs = require("fs").promises;
+const path = require("node:path");
+const { nanoid } = require("nanoid");
 
-const getContactById = async (contactId) => {}
+const contactsPath = path.join(__dirname, "contacts.json");
 
-const removeContact = async (contactId) => {}
+const updateContactsFile = async (writtenData) => {
+  await fs.writeFile(contactsPath, JSON.stringify(writtenData));
+};
 
-const addContact = async (body) => {}
 
-const updateContact = async (contactId, body) => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath);
+  const parseData = JSON.parse(data);
+  return parseData;
+};
+
+const getContactById = async (contactId) => {
+  const data = await listContacts();
+  const selectedContact = data.find((el) => el.id === contactId);
+  return selectedContact;
+};
+
+
+const addContact = async (name, email, phone) => {
+  const data = await listContacts();
+  const addedContact = {
```

Comment:
teoretycznie to jeszcze nie jest dodany contact lepiej byłoby contact albo contactToAdd